lib/src/app/provisioning.dart

```diff
@@ -0,0 +1,51 @@
+import 'dart:async';
+
+import '../gen/provisioning/v1/provisioning.pbgrpc.dart';
+
+/// gRPC client for connecting to Viam's Provisioning Service
+///
+/// All calls must be authenticated.
+class ProvisioningClient {
+  final ProvisioningServiceClient _client;
+
+  ProvisioningClient(this._client);
+
+  /// Get the status of the SmartMachine
+  Future<GetSmartMachineStatusResponse> getSmartMachineStatus() async {
+    final request = GetSmartMachineStatusRequest();
+    final response = await _client.getSmartMachineStatus(request);
+    return response;
+  }
+
+  /// Set the network credentials of the Smart machine, so it can connect to the internet.
+  ///
+  /// [type] is either 'wired' or 'wifi'
```

Comment:
Maybe make this an `enum` to make it strongly typed?

Reply:
good point, i thought about that but skipped for now. Will circle back and add in. Thanks!